### ISSUE TITLE
Ariba fails when using virulencefinder db

### ISSUE THREAD

**Issue by Csekse** [sanger-pathogens/ariba#216]
Thanks for a nice programme.
We have used Ariba With both Card and Resfinder database and it Works well.
Today I have also tested the virulencefinder db and get the following error Message (see below).
I've also run With resfinder today as well and it Works fine.
Do you have any idea what's going wrong? And if there is there anything you could do to help I would be very Grateful.
Best regards, Camilla 

[E::hts_idx_push] Unsorted positions on sequence #1: 16 followed by 2
Traceback (most recent call last):
  File "/work/projects/nn9305k/src/anaconda3/bin/ariba", line 298, in <module>
    args.func(args)
  File "/work/projects/nn9305k/src/anaconda3/lib/python3.6/site-packages/ariba/tasks/run.py", line 65, in run
    c.run()
  File "/work/projects/nn9305k/src/anaconda3/lib/python3.6/site-packages/ariba/clusters.py", line 602, in run
    self._run()
  File "/work/projects/nn9305k/src/anaconda3/lib/python3.6/site-packages/ariba/clusters.py", line 627, in _run
    self._init_and_run_clusters()
  File "/work/projects/nn9305k/src/anaconda3/lib/python3.6/site-packages/ariba/clusters.py", line 424, in _init_and_run_clusters
    reference_names=self.cluster_ids[cluster_name],
KeyError: ''

**Comment by Asutu** [comment]
I'm getting exactly the same error with virulencefinder. Any idea what the problem is and/or how to solve it?

Cheers,
Pedro

**Comment by martinghunt** [comment]
Thanks for reporting this @Csekse and @Asutu.

After a little digging I found a badly formatted FASTA file in the virulencefinder download. ariba will fix it when running `getref`, so you'll need to rerun `getref` and `prepareref`. The fix is in the new release 2.11.2. Could you please try it? I am hoping that it will stop the crashes that you found on your data.

**Comment by Asutu** [comment]
Thanks for trying to sort things out. Unfortunately, I'm still getting the same error message as above with Ariba 2.11.2.

 ```
 [E::hts_idx_push] Unsorted positions on sequence #1: 1009 followed by 101
  Traceback (most recent call last):
    File "/scratch/scratch/ucjucoe/Software/bacterialGenomics/ariba/bin/ariba", line 298, in <module>
      args.func(args)
    File "/scratch/scratch/ucjucoe/Software/bacterialGenomics/ariba/lib/python3.6/site-packages/ariba/tasks/run.py", line 65, in run
      c.run()
    File "/scratch/scratch/ucjucoe/Software/bacterialGenomics/ariba/lib/python3.6/site-packages/ariba/clusters.py", line 602, in run
      self._run()
    File "/scratch/scratch/ucjucoe/Software/bacterialGenomics/ariba/lib/python3.6/site-packages/ariba/clusters.py", line 613, in _run
      self._map_and_cluster_reads()
    File "/scratch/scratch/ucjucoe/Software/bacterialGenomics/ariba/lib/python3.6/site-packages/ariba/clusters.py", line 276, in _map_and_cluster_reads
      log_fh=filehandle
    File "/scratch/scratch/ucjucoe/Software/bacterialGenomics/ariba/lib/python3.6/site-packages/ariba/read_store.py", line 19, in __init__
      self._compress_and_index_file(self.outprefix, log_fh)
    File "/scratch/scratch/ucjucoe/Software/bacterialGenomics/ariba/lib/python3.6/site-packages/ariba/read_store.py", line 35, in _compress_and_index_file
      pysam.tabix_index(infile + '.gz', seq_col=0, start_col=1, end_col=1)
    File "pysam/libctabix.pyx", line 1025, in pysam.libctabix.tabix_index
  OSError: building of index for virulencefinder/read_store.gz failed
```
Here's the output of running `ariba version`:

```
ARIBA version: 2.11.2

External dependencies:
bowtie2	2.3.2	/home/pmca/Software/ngs/bowtie2/2.3.2-legacy/bowtie2
cdhit	4.7	/home/pmca/Software/bio-misc/cd-hit/4.6.8/cd-hit-est
nucmer	3.1	/home/pmca/Software/alignments/mummer/3.23-64bits/nucmer
spades	3.11.1	/home/pmca/Software/assemblers/SPAdes/SPAdes-3.11.1-Linux/bin/spades.py

External dependencies OK: True

Python version:
3.6.3 (default, Oct 10 2017, 15:11:55) 
[GCC 4.9.2]

Python packages:
ariba	2.11.2	/home/pmca/Software/bacterialGenomics/ariba/lib/python3.6/site-packages/ariba/__init__.py
bs4	4.6.0	/home/pmca/Software/bacterialGenomics/ariba/lib/python3.6/site-packages/bs4/__init__.py
dendropy	4.3.0	/home/pmca/Software/bacterialGenomics/ariba/lib/python3.6/site-packages/dendropy/__init__.py
pyfastaq	3.17.0	home/pmca/Software/bacterialGenomics/ariba/lib/python3.6/site-packages/pyfastaq/__init__.py
pymummer	0.10.3	/home/pmca/Software/bacterialGenomics/ariba/lib/python3.6/site-packages/pymummer/__init__.py
pysam	0.14.1	/home/pmca/Software/bacterialGenomics/ariba/lib/python3.6/site-packages/pysam/__init__.py

Python packages OK: True

Everything looks OK: True
```

**Comment by martinghunt** [comment]
@Asutu could you try the latest release 2.12.0?

**Comment by ys4** [comment]
@martinghunt It appears that the getref command in 2.12.0 release

`ariba getref virulencefinder out.virulencefinder`

points to the plasmidfinder data instead.

**Comment by martinghunt** [comment]
@ys4 thank you for reporting! It's fixed in the new release 2.12.1.
